### PR TITLE
[discreture] Add new port

### PR DESCRIPTION
**Pull request by remz1337**
Add new port. There is one small problem at the end of the installation, the command vcpkg gives to include the port in your CMakeLists file points to the wrong header file. It says to add Discreture/ArithmeticsProgression.hpp, but it should be discreture.hpp.

How do I fix this?

### CODE REVIEW

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/discreture/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    REF eeeec31c814e6a9a8506a6bfd6a5b35704350605
+    SHA512 de1c7d74d337605fd9b9d1f3ee6637b4afd179d495de243b21168b0a4376b83c0519b4cced985af694850755ab1e3caca5087b3ca0cd6ccb3b73b10bd6b25b49
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -Ddiscreture_INSTALL_CMAKE_DIR=share
```
[the remark this answers is not in the document]
Comment (remz1337) [reply ports/discreture/portfile.cmake]: Sorry, I copied the portfile from another package and didn't check this. Removed in new commit

File: ports/discreture/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+vcpkg_install_cmake()
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug
+    ${CURRENT_PACKAGES_DIR}/share/discreture-config-version.cmake)
```
[the remark this answers is not in the document]
Comment (remz1337) [reply ports/discreture/portfile.cmake]: Removed in latest commit

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Hi @remz1337 

Please `vcpkg x-add-version discreture` and submit the changes to solve the failures on x86-windows.
```
Found the following errors:
Error: While validating port discreture.
       Missing expected versions file at: C:\a\1\s\versions\d-\discreture.json
       Run:

           vcpkg x-add-version discreture

       to create the versions file.
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by remz1337** [comment]
Thank you @NancyLi1013,
I couldn't find anything in the docs about how to setup correctly the `find_path ` command that vcpkg shows at the end of the installation, do you know where I could find more information or maybe an example? Or is it something managed by the port itself?

**Comment by strega-nil** [comment]
@remz1337 `usage` files are how one does this, for example, <https://github.com/microsoft/vcpkg/blob/master/ports/caf/usage>. You can then just copy that file over to the installation directory :)

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**